source/common/upstream/load_balancer_impl.cc

```diff
@@ -773,6 +831,44 @@ void EdfLoadBalancerBase::refresh(uint32_t priority) {
         HostsSource(priority, HostsSource::SourceType::LocalityDegradedHosts, locality_index),
         host_set->degradedHostsPerLocality().get()[locality_index]);
   }
+  if (slow_start_enabled_) {
+    recalculateHostsInSlowStart(host_set->hosts());
+  }
+}
+
+bool EdfLoadBalancerBase::adheresToEndpointWarmingPolicy(const Host& host) {
+  // Enter slow start innediately.
+  if (host.healthChecker().isNull()) {
```

Comment:
am now thinking that simplier way could have been just to check `host.health() == Upstream::Host::Health::Healthy`. It works for both cases, if no HC is configured then host is always healthy otherwise active HC would update host health flag accordingly.